### DISCUSSION TITLE
Is There a Recommended Method to Reset the Admin Password When It's Forgotten in last version?

### DISCUSSION THREAD

**Discussion by ehsanr1366** [keycloak/keycloak#34600] [Q&A]
In cases where the Keycloak admin password is forgotten, and no reset mechanisms have been preconfigured, is there an official or recommended way to reset it? Since access to the Keycloak UI would not be possible under these circumstances, it would be helpful to have guidance on any CLI-based or alternative recovery methods.

Thank you for any suggestions or best practices for this scenario.

**Comment by sschu** [comment]
https://www.keycloak.org/server/bootstrap-admin-recovery

**Reply by ehsanr1366** [reply]
Thanks. I can not do it.

./kc.sh show-config
Current Mode: development
Current Configuration:
```
        kc.bootstrap.admin.password =  temp_admin (KcEnvVarConfigSource)
        kc.bootstrap.admin.username =  temp_admin (KcEnvVarConfigSource)
```


But:
`./kc.sh start-dev
`
```
./kcadm.sh config credentials --server http://localhost:8080 --realm master --user temp_admin --password "temp_admin"
Logging into http://localhost:8080 as user temp_admin of realm master
Invalid user credentials [invalid_grant]
```


and:

```
./kc.sh bootstrap-admin user
Unknown option: 'bootstrap-admin'
Did you mean: kc.sh tools or kc.sh start-dev?
```

**Reply by sschu** [reply]
Which Keycloak version are you using? I just tried with KC26 and it worked like a charm.

**Reply by heruan** [reply]
> Which Keycloak version are you using? I just tried with KC26 and it worked like a charm.

Hey @sschu when I run `./kc.sh bootstrap-admin user --username admin --password foobar` I get:
```
ERROR: duplicate key value violates unique constraint "uk_ru8tt6t700s9v50bu18ws5ha6"
```
Since it's trying to `insert` the `admin` user again, instead of resetting the password.

**Reply by sschu** [reply]
The command is not designed to reset the password for existing users, only to create new ones: https://www.keycloak.org/server/bootstrap-admin-recovery

**Reply by heruan** [reply]
Thanks for the clarification! I've read through the docs, looks like there's no support for an admin password reset, is that the case? Also looks like the server must be stopped for the command to exit cleanly. What would be the best strategy to reset an admin password via CLI, on a running instance?

**Reply by sschu** [reply]
There is no supported option I know of.  Maybe you can copy over a known password from a different user on the database directly if nothing else works for you.
